web/src/shared/components/pedigree/TangledTree.tsx

```diff
@@ -11,19 +15,672 @@ interface PedigreeEntry {
+                            parentsList: [
+                                keyedData[next].paternal_id,
+                                keyedData[next].maternal_id,
+                            ].filter((l) => l),
+                        }),
+                    },
+                ]
+                yetToSee.delete(next)
+                updatedList = true
+                break
+            }
+            // try add child below parents
+            const parentLevel = max([
+                findInHeirarchy(keyedData[next].maternal_id, toReturn),
+                findInHeirarchy(keyedData[next].paternal_id, toReturn),
+            ])
+            if (parentLevel > -1) {
+                toReturn[parentLevel + 1] = [
+                    ...toReturn[parentLevel + 1],
+                    {
+                        id: next,
+                        ...((keyedData[next].paternal_id || keyedData[next].maternal_id) && {
+                            parentsList: [
+                                keyedData[next].paternal_id,
+                                keyedData[next].maternal_id,
+                            ].filter((m) => m),
+                        }),
+                    },
+                ]
+                yetToSee.delete(next)
+                updatedList = true
+                break
+            }
+        }
+        if (!updatedList) {
+            break
+        }
+        updatedList = false
+    }
+
+    const nodeDefaults = {
+        x: 0,
+        y: 0,
+        parents: [],
+        level: -1,
+        bundles: [],
+        links: [],
+        parentsList: [],
+    }
+
+    return toReturn.map((l) => ({
+        nodes: l.map((m) => ({ ...nodeDefaults, ...m })),
+        bundles: [],
+    }))
+}
+
 interface RenderPedigreeProps {
     data: PedigreeEntry[]
     click?(e: string): void
 }
 
-const TangledTree: React.FunctionComponent<RenderPedigreeProps> = ({ data, click }) => (
-    <>
-        Temporary Pedigree component
-        <br />
-        {JSON.stringify(data)}
-        <br />
-        {JSON.stringify(click)}
-    </>
-)
+const TangledTree: React.FunctionComponent<RenderPedigreeProps> = ({ data, click }) => {
+    const [tree, setTree] = React.useState<NodeList[]>([])
+    const [keyedData, setKeyedData] = React.useState<{
+        [name: string]: PedigreeEntry
+    }>()
+    const onClick =
+        click ??
+        function defaultClick() {
+            return undefined
+        }
+
+    React.useEffect(() => {
+        setTree(formatData(data))
+        setKeyedData(
+            data.reduce((obj: { [key: string]: PedigreeEntry }, s: PedigreeEntry) => {
+                obj[s.individual_id] = s
+                return obj
+            }, {})
+        )
+    }, [data])
+
+    return (
+        <>
+            {(!data || !data.length) && (
+                <MuckError message={`Ah Muck, there's no data for this pedigree!`} />
+            )}
+            {data && !!data.length && !tree && (
```

Comment:
I think it would be better if you handle the loading and error states outside of this component, and let this responsibility be on the parent component, for example:

```es6
# In parent component

if (error) return <MuckError message={error.message} />
if (loading) return <LoadingDucks />

const pedigreeData = data?.family.project.pedigree
if (!pedigreeData) {
  return <div>Pedigree data not available</div>
} else {
  return <TangledTree data={pedigreeData} click={onClick} />
}
```

Then I would delete most of the rendering logic here and simply have:

```es6
# In this component

if (tree && keyedData) {
  return renderChart(tree, keyedData, onClick)
} else {
  return <div>No pedigree to show</div>
}
```